builtin: add null checks to TileEnvelope function

cty123 opened the pull request:
The existing `TileEnvelope()` builtin function expects the passed in `bounds` object to have valid `BoundingBoxRef` object, and the nil check is missing, such that certain input can cause server panic. This MR simply adds the nil check to error out when the input is invalid.

Fixes: #115405
Release note: None

cockroach-teamcity commented:
[![CLA assistant check](https://cla.crdb.dev/pull/badge/not_signed)](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=115494) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=115494) before we can accept your contribution.<br/><hr/>**Tianyu Chen** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.crdb.dev/check/cockroachdb/cockroach?pullRequest=115494) it.</sub>

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/115494)
<!-- Reviewable:end -->

cty123 committed:
builtin: add null checks to TileEnvelope function

The existing `TileEnvelope()` builtin function expects the passed in `bounds` object to have valid `BoundingBoxRef` object, and the nil check is missing, such that certain input can cause server panic. This MR simply adds the nil check to error out when the input is invalid.

Fixes: #115405
Release note: None

yuzefovich commented:
cc @rharding6373
